### ISSUE TITLE
Settings Lost After Client Update

### ISSUE THREAD

**Issue by ekianjo** [ValveSoftware/SteamOS#554]
#### Your system information

* Steam client version: latest stable
* SteamOS version: 2.87
* Opted into Steam client beta?: No 
* Opted into SteamOS beta?: No 
* Have you checked for updates in Settings > System?: Yes

#### Please describe your issue in as much detail as possible:
Simple. Whenever the client updates at the start of the Steam Machine, there is a good probability (not 100%, but maybe once out of 2) that the settings are completely lost are need to be re-entered. Such as, the login users (previous registered users are completely erased, need to start from scratch, which is time consuming if you have several users to set up), and all settings in the BPM client options are gone, too. Such as sound settings (need to resetup the audio output), resolution (need to adjust my monitor display again), desktop settings (need to ensure I can go into desktop linux again), etc etc... this is pretty annoying when it keeps happening over and over again. I'm not sure what is causing this issue, but it would be great if the update process could avoid erasing such settings or back them up before updating in case they are lost.

#### Steps for reproducing this issue:

1. Upgrade to new Steam client at boot
2. May not happen every single time but often enough to be reproducible.

**Comment by tiberiusferreira** [comment]
Does it still happen with 2.88?

**Comment by ekianjo** [comment]
Is 2.88 stable yet ? My understanding is that it's still in beta right?

**Comment by tiberiusferreira** [comment]
Yes, it is. I just asked because maybe it was resolved in the latest beta.

**Comment by ekianjo** [comment]
@tiberiusferreira I don't test SteamOS beta, sick already of always having to reinstall the stable one without having to worry about beta bugs :P But i dont think that issue was mentioned as being worked on for 2.88 unless I missed something.

**Comment by ProfessorKaos64** [comment]
I have not updated yet to the OS beta 2.88, but I can confirm that a Steam client update acts as if my Steam Controller was not previously registered to my account.

**Comment by ekianjo** [comment]
Happened to me again yesterday. I see some pattern... first at boot I got a update of 3x Mbs, followed immediately after by a 24x Mb update... and another 3x Mb update - as if it was re-dowloading the whole client, then the update of the client once more. Really wierd behavior. Then I got the login with no users registered at all, and had to recreate them again. 

Note that when I get such an issue, I also get stuck once into "Updating Steam User xxx" loop that never ends (no progress bar displays) making me have to reboot once more to go past that stage.

**Comment by CrazySpence** [comment]
No ones updated this in awhile but I also have this issue fairly frequently.

-I'll get the update notification in the upper right
-I'll reboot to do the update
-Sometimes ill have to log in again completely (this happens the least amount but it has happened several times)
-Most of the time I have to reconfigure audio
-Most of the time I have to reconfigure non steam gamepads again
-I'll also have to reconfigure the "switch to desktop mode" option
